File: engine/runtime-compiler/src/main/scala/org/enso/compiler/CompilerResult.scala
package org.enso.compiler

import org.enso.compiler.context.CompilerContext.Module

/** The result of running the compiler.
  *
  * @param compiledModules the modules compiled during the run
  */
final case class CompilerResult(compiledModules: List[Module])
object CompilerResult {

  /** @return an empty compiler result */
  def empty: CompilerResult =
    CompilerResult(Nil)
}

File: engine/runtime-compiler/src/main/java/org/enso/compiler/CompilerResult.java
package org.enso.compiler;

import org.enso.compiler.context.CompilerContext.Module;
import org.enso.scala.wrapper.ScalaConversions;
import scala.collection.immutable.List;

/**
 * The result of running the compiler.
 *
 * @param compiledModules the modules compiled during the run
 */
public record CompilerResult(List<Module> compiledModules) {
  /**
   * @return an empty compiler result
   */
  public static CompilerResult empty() {
    return new CompilerResult(ScalaConversions.nil());
  }
}
